### ISSUE TITLE
Undocumented dependency on JQuery?

### ISSUE THREAD

**Issue by davidmegginson** [okfn/ckan.js#22]
Readme.md doesn't mention a JQuery dependency, but the library is failing when I don't have JQuery loaded.

**Comment by rgrp** [comment]
@davidmegginson are you using this from browser or node? In node should not require jquery.

**Comment by davidmegginson** [comment]
@rgrp I was using it in-browser, for a client-only app.

**Comment by rgrp** [comment]
@davidmegginson that's definitely a thought :-) - only trouble is that can be a little bit painful (e.g. if you need to add auth headers etc).

What's definitely missing right now is a something in the README about needing jquery - if you have a suggestion there please open a pull request or we can add it at our end :-)

**Comment by neothemachine** [comment]
I don't think this library should depend on jquery. Maybe there are smaller libraries made just for that purpose (AJAX).

**Comment by loleg** [comment]
I agree with @letmaik - and in today's PR #34 proposed to use the lightweight needle-js library. Can we completely remove the need for jQuery, perhaps even abstract the Ajax call to allow the user to use whatever is already part of their stack?

**Comment by rufuspollock** [comment]
@loleg yes we can completely remove and also happy for us to use what is in people's stack - if there is a clean way to do that.

**Comment by loleg** [comment]
On further thought, [superagent](https://github.com/visionmedia/superagent/blob/master/test/support/express/responseDecorator.js#L230) is the only modern library with JSONP support (albeit undocumented), the reason I've been continuing to use jQuery in [ckan-embed](https://github.com/opendata-swiss/ckan-embed#advanced-options). 

Would it potentially make sense to abstract away the HTTP request, so the library user can plug in whatever they want, with perhaps just a sensible default? If anyone with experience in the various options out there can weigh in here, please do.

**Comment by rufuspollock** [comment]
@loleg happy for you to lead here and make a proposal - if superagent seems good to you happy to go for that.
